### PR TITLE
Add export module

### PR DESCRIPTION
**Pull request by ckanesan**
Adds a module that exports the model to the ONNX framework and executes a test run. 
Some minor modifications are made to the model to make it exportable but should not affect the outputs.

### CODE REVIEW

File: hawp/detector.py [hunk 1]
```diff
@@ -120,14 +120,18 @@ def forward(self, images, annotations = None):
         #jloc_pred_nms = non_maximum_suppression(jloc_pred[0])
         #topK = torch.clamp((jloc_pred_nms > 0.008).count_nonzero(), max=300)
 
-        juncs_pred, _ = get_junctions(non_maximum_suppression(jloc_pred[0]),joff_pred[0], topk=300, th=0.008)
+        nms_jloc_pred = non_maximum_suppression(jloc_pred)[0]
```
Comment (GrimReaperSam) [inline hawp/detector.py:120]: This is the part where I pass the 4 dimensional tensor, and then take out the batch dim after max_pool2d

File: hawp/detector.py [hunk 2]
```diff
@@ -120,14 +120,18 @@ def forward(self, images, annotations = None):
...
+        juncs_pred, _ = get_junctions(nms_jloc_pred,joff_pred[0], topk=300, th=0.008)
         extra_info['time_proposal'] = time.time() - extra_info['time_proposal']
         extra_info['time_matching'] = time.time()
         dis_junc_to_end1, idx_junc_to_end1 = torch.sum((lines_pred[:,:2]-juncs_pred[:,None])**2,dim=-1).min(0)
         dis_junc_to_end2, idx_junc_to_end2 = torch.sum((lines_pred[:,2:] - juncs_pred[:, None]) ** 2, dim=-1).min(0)
 
-        idx_junc_to_end_min = torch.min(idx_junc_to_end1,idx_junc_to_end2)
-        idx_junc_to_end_max = torch.max(idx_junc_to_end1,idx_junc_to_end2)
+        # idx_junc_to_end_min = torch.min(idx_junc_to_end1,idx_junc_to_end2)
+        # idx_junc_to_end_max = torch.max(idx_junc_to_end1,idx_junc_to_end2)
```
Comment (GrimReaperSam) [inline hawp/detector.py:128]: torch.min and torch.max signatures are (tensor, dim, keep_dims, ..)
They are used here are torch.min/max(tensor1, tensor2) to do element-wise minimum/maximum operations.
This does not work for onnx, as it sees min and max, and expects the second argument to be an integer, and not a tensor.

The fix is to stack the two tensors together, then take the min/max along the stacked dimension.

File: hawp/detector.py [hunk 3]
```diff
@@ -155,10 +159,13 @@ def forward(self, images, annotations = None):
...
             edge_indices = torch.stack((v2e_idx1,v2e_idx2)).t()
-            wireframe = WireframeGraph(juncs_final,juncs_score,edge_indices,score_final,output.size(3),output.size(2))
 
-            wireframe.rescale(annotations[0]['width'],annotations[0]['height'])
-        
+            if not self.export_mode:
+                wireframe = WireframeGraph(juncs_final,juncs_score,edge_indices,score_final,output.size(3),output.size(2))
+                wireframe.rescale(annotations[0]['width'],annotations[0]['height'])
+            else:
+                return juncs_final,juncs_score,edge_indices,score_final,output.size(3),output.size(2)
```
Comment (GrimReaperSam) [inline hawp/detector.py:164]: output.size(3) and output.size(2) are not tensors anymore, they probably won't have meaningful data once exported
